projet/users/views.py

```diff
@@ -94,8 +103,59 @@ class ListUsers(generics.ListAPIView):
     to add another permissions apart from the IsAuthenticated
     """
 
+    search_fields = ["email", "first_name", "last_name"]
+    filter_backends = (filters.SearchFilter,)
+
     queryset = User.objects.all()
-    serializer_class = UserSerializer
+    serializer_class = UserBulkSerializer
+
+
+def send_invite(project, recipient, sender=None):
+    send_mail(
+        "Projet",
+        f"You have been asked by {sender.email} to contribute to {project.title}. Accept at 127.0.0.1:8000/{project.id}/",
+        recipient_list=[recipient.email],
+        from_email=settings.EMAIL_HOST,
+    )
+
+    Contributor.objects.create(user=recipient, project=project)
+    return "Email sent"
+
+
+@api_view(["POST"])
+def invite(request):
```

Comment:
The invite view.
Later we'll clean up this view and put the permissions in a separate file.